### ISSUE TITLE
Patching `ghcup` for gold linker can be avoided by only using `LD=ld.gold`

### ISSUE THREAD

**Issue by evolutics** [jkachmar/alpine-haskell-stack#4]
The file [`docker/ghcup.diff`](https://github.com/jkachmar/alpine-haskell-stack/blob/master/docker/ghcup.diff) patches `ghcup` to make it work with the gold linker.

I may be missing something, but instead of the patch it is enough to change the call `ghcup compile …` to `LD=ld.gold ghcup compile …`. The option [`--disable-ld-override` for `configure`](https://gitlab.haskell.org/ghc/ghc/blob/master/aclocal.m4#L2403) should not be required, as GHC by default uses the linker from the `LD` environment variable.

With this simplification, `make docker-build-gmp` still works fine for me.

The diff is as follows:

```diff
diff --git a/Dockerfile b/Dockerfile
index 85caa1c..74c408e 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -33,11 +33,6 @@ RUN apk upgrade --no-cache &&\
 
 COPY docker/ghcup /usr/bin/ghcup
 
-# Workaround hardcoded linker settings in GHC's build by patching 'ghcup'
-COPY docker/ghcup.diff /tmp/ghcup.diff
-RUN patch /usr/bin/ghcup < /tmp/ghcup.diff &&\
-    rm /tmp/ghcup.diff
-
 ################################################################################
 # Intermediate layer that builds GHC
 FROM base AS build-ghc
@@ -78,7 +73,7 @@ RUN if [ "${GHC_BUILD_TYPE}" = "gmp" ]; then \
 fi
 
 RUN echo "Compiling and installing GHC" &&\
-    ghcup -v compile -j $(nproc) -c /tmp/build.mk ${GHC_VERSION} ghc-8.4.3 &&\
+    LD=ld.gold ghcup -v compile -j $(nproc) -c /tmp/build.mk ${GHC_VERSION} ghc-8.4.3 &&\
     rm /tmp/build.mk &&\
     echo "Uninstalling GHC bootstrapping compiler" &&\
     apk del ghc &&\
diff --git a/docker/ghcup.diff b/docker/ghcup.diff
deleted file mode 100644
index 26cbf27..0000000
--- a/docker/ghcup.diff
+++ /dev/null
…
```

**Comment by symbiont-joseph-kachmar** [comment]
This looks good. I think the patch was a carryover from wherever I adapted this work from. Thanks for the issue, I'll make a note to change this in the repo!

**Comment by jkachmar** [comment]
Addressed by c637bc74e9b099da23e37dfba799c93443f465b7; thanks again!
